cdap-authorization-dataset-plugin/src/main/java/co/cask/cdap/security/authorization/ACLDataset.java

```diff
@@ -115,6 +116,28 @@ public void remove(EntityId entity) {
     }
   }
 
+  public Set<Privilege> listPrivileges(Principal principal) {
+    Set<Privilege> privileges = new HashSet<>();
+    // scan the whole table
+    Scanner scan = table.scan(null, null);
```

Comment:
Generally when you want to do forward and reverse look-ups, you'll have to store both forward and reverse mapping. This also brings up another question, I don't see any APIs to remove ACLs for a principal.